app/src/main/java/com/del/delcontainer/ui/settings/SettingsViewModel.java

```diff
@@ -1,19 +1,61 @@
 package com.del.delcontainer.ui.settings;
 
+import android.util.Log;
+
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
+import com.del.delcontainer.database.entities.UserProfile;
+import com.del.delcontainer.repositories.UserProfileRepository;
+import com.del.delcontainer.ui.login.LoginStateRepo;
+import com.del.delcontainer.utils.apiUtils.APIUtils;
+import com.del.delcontainer.utils.apiUtils.interfaces.UserApi;
+import com.del.delcontainer.utils.apiUtils.pojo.UserDetails;
 
-public class SettingsViewModel extends ViewModel {
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
 
-    private MutableLiveData<String> mText;
+public class SettingsViewModel extends ViewModel {
 
-    public SettingsViewModel() {
-        mText = new MutableLiveData<>();
-        mText.setValue("User Name");
+    private static final String TAG = "SettingsViewModel";
+    private MutableLiveData<String> firstName = new MutableLiveData<>();
+    private MutableLiveData<LoginStateRepo> loginStateRepo = new MutableLiveData<>();
+    public MutableLiveData<LoginStateRepo> getLoginStateRepo() {
+        return loginStateRepo;
     }
 
-    public LiveData<String> getText() {
-        return mText;
+    Retrofit retrofit = APIUtils.getApiClient();
+
+    /**
+     * Fetch first name of the user
+     *
+     * @param token
+     * @param userId
+     */
+    public void getUserFirstName(String token, String userId){
```

Comment:
Why just the first name? Are there any other details that you can fetch?

Reply:
Yes, only first name for now. I can similarly fetch other details, but I didn’t know exactly where to show them. Showcasing them in a separate fragment or dialogue would take more hours of setting up UI so I couldn’t find doing that after Wednesday shift. Will start soon.